tests/test_user_api.py

```diff
@@ -0,0 +1,329 @@
+                url = furl(f'/v1/users/{test["name"]}/groups/')
+                query_params = {
+                    'user_id': test['name'],
+                    'action': test['action']
+                }
+                url.add(query_params=query_params)
+                user = User.provision_user(directory, test['name'])
+                if test['action'] == 'remove':
+                    user.add_groups(test['json_request_body']['groups'])
+                resp = self.app.put(url.url, headers=headers, data=data)
+                self.assertEqual(test['response']['code'], resp.status_code)
+                resp.raise_for_status()
+
+    def test_get_username_groups(self):
+        headers = {'Content-Type': "application/json"}
+        headers.update(get_auth_header(service_accounts['admin']))
+        name = "test_user_group_api@email.com"
+        user = User.provision_user(directory, name)
+        resp = self.app.get(f'/v1/users/{name}/groups', headers=headers)
+        self.assertEqual(0, len(json.loads(resp.body)['groups']))
+        user.add_groups([Group.create(directory, "group_0").name, Group.create(directory, "group_1").name])
+        resp = self.app.get(f'/v1/users/{name}/groups', headers=headers)
+        self.assertEqual(2, len(json.loads(resp.body)['groups']))
+
+        resp.raise_for_status()
+
+    def test_put_username_roles(self):
+        tests = [
+            {
+                'name': "test_put_user_role0@email.com",
+                'action': 'add',
+                'json_request_body': {
+                    "roles": [Role.create(directory, "role_0").name]
+                },
+                'response': {
+                    'code': 200
+                }
+            },
+            {
+                'name': "test_put_user_role1@email.com",
+                'action': 'remove',
+                'json_request_body': {
+                    "roles": [Role.create(directory, "role_1").name]
+                },
+                'response': {
+                    'code': 200
+                }
+            }
+        ]
+        for test in tests:
+            with self.subTest(test['json_request_body']):
+                data = json.dumps(test['json_request_body'])
+                headers = {'Content-Type': "application/json"}
+                headers.update(get_auth_header(service_accounts['admin']))
+                url = furl(f'/v1/users/{test["name"]}/roles/')
+                query_params = {
+                    'user_id': test['name'],
+                    'action': test['action']
+                }
+                url.add(query_params=query_params)
+                user = User.provision_user(directory, test['name'])
+                if test['action'] == 'remove':
+                    user.add_roles(test['json_request_body']['roles'])
+                resp = self.app.put(url.url, headers=headers, data=data)
+                self.assertEqual(test['response']['code'], resp.status_code)
```

Comment:
if you're checking the response code here you don't need `resp.raise_for_status()` below